File: Python/백준/Bronze/8595. 히든 넘버/히든 넘버.py
n = int(input())
s = input()

current_number = 0
total_sum = 0

for char in s:
    if char.isdigit(): # 주어진 문자가 숫자인지 확인하는 조건
        current_number = current_number * 10 + int(char) # 연속된 숫자 만들어 주기
    else:
        total_sum += current_number # 합계
        current_number = 0

total_sum += current_number

print(total_sum)

File: Java/백준/Bronze/8595. 히든 넘버/히든 넘버.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        String s = br.readLine();

        long totalSum = 0;
        long currentNumber = 0;

        for (char c : s.toCharArray()) {
            if (Character.isDigit(c)) {  // 숫자인 경우
                currentNumber = currentNumber * 10 + (c - '0');  // 숫자 조합(확장)
            } else {
                totalSum += currentNumber;
                currentNumber = 0;
            }
        }

        totalSum += currentNumber;

        System.out.println(totalSum);

        br.close();
    }
}
